pkg/upstream/write.go

```diff
@@ -100,7 +115,25 @@ func WriteUpstream(u *types.Upstream, options types.WriteOptions) error {
 			u.Files[i] = file
 		}
 
-		if err := ioutil.WriteFile(fileRenderPath, file.Content, 0644); err != nil {
+		if archives.IsTGZ(file.Content) {
+			// this is a helm chart, so we need to check if it is or contains the replicated-sdk
+			reader := bytes.NewReader(file.Content)
+			replicatedSDKChartName, isReplicatedSDK, err := FindReplicatedSDKChart(reader, u.ReplicatedChartNames)
+			if err != nil {
+				return errors.Wrap(err, "failed to find replicated-sdk subchart")
+			}
+
+			if replicatedSDKChartName != "" {
```

Comment:
why only if it's a subchart?

Reply:
this should have a value if it's found at all, regardless of whether it's a parent or subchart. the `isReplicatedSDK` variable (now renamed to `isSubchart`) is what makes the distinction